src/status_im/contexts/wallet/tokens/events.cljs

```diff
@@ -20,38 +20,48 @@
-                                        (tokens-data/tokens-by-symbol
-                                         {:added-tokens by-symbol
-                                          :source-name  name
-                                          :tokens       tokens
-                                          :chain-ids    chain-ids}))))
-                          {:sources    []
-                           :by-address {}
-                           :by-symbol  {}}
-                          data)
-        symbols   (->> tokens
-                       :by-symbol
-                       keys
-                       (remove utils.address/address?))]
+  (let [chain-ids                  (chain/chain-ids db)
+        tokens                     (reduce (fn [{:keys [by-address by-symbol] :as data}
+                                                {:keys [name source version tokens]}]
+                                             (-> data
+                                                 (update :sources
+                                                         conj
+                                                         {:name         name
+                                                          :source       source
+                                                          :version      version
+                                                          :tokens-count (count tokens)})
+                                                 (update :by-address
+                                                         merge
+                                                         (tokens-data/tokens-by-address
+                                                          {:added-tokens by-address
+                                                           :source-name  name
+                                                           :tokens       tokens
+                                                           :chain-ids    chain-ids}))
+                                                 (update :by-symbol
+                                                         merge
+                                                         (tokens-data/tokens-by-symbol
+                                                          {:added-tokens by-symbol
+                                                           :source-name  name
+                                                           :tokens       tokens
+                                                           :chain-ids    chain-ids}))))
+                                           {:sources    []
+                                            :by-address {}
+                                            :by-symbol  {}}
+                                           data)
+        symbols                    (->> tokens
+                                        :by-symbol
+                                        keys
+                                        (remove utils.address/address?))
```

Comment:
We recently made a refactor on this here #21453, so make sure to include it when rebasing